.pre-commit-config.yaml

```diff
@@ -17,6 +17,12 @@ repos:
       - id: mixed-line-ending
         name: "Normalize mixed line endings"
         args: [--fix=lf]
+      - id: end-of-file-fixer
+        name: "Ensure files end with a single newline"
+      - id: trailing-whitespace
+        name: "Remove trailing whitespace"
+        args: ["--chars", " \t"]
```

Comment:
Why only trim trailing spaces and tabs? What trailing whitespace do we want to conserve? 

https://github.com/pre-commit/pre-commit-hooks?tab=readme-ov-file#trailing-whitespace

Reply:
Form-feed characters before the emacs stanza in older PEPs, mainly. I'd want to remove the stanzas and the form-feed characters at the same time.

Reply:
Shall we do it in this PR? How many extra files and pings would it be?

Reply:
~290 files (searching for " Local Variables:") Currently this PR touches 90 files.

Reply:
And 37 -> 81 pings...

Reply:
My thinking with #4081 was to include 'everything' in one PR, so we don't do a mass ping again (or, try and minimise the number of mass pings). It's a fine line, though.